pkg/util/dynamichelper/dynamichelper_test.go

```diff
@@ -4,17 +4,92 @@ package dynamichelper
 // Licensed under the Apache License 2.0.
 
 import (
+	"context"
+	"errors"
 	"reflect"
 	"testing"
 	"time"
 
+	"github.com/golang/mock/gomock"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 
 	"github.com/Azure/ARO-RP/pkg/util/cmp"
+	mock_dynamichelper "github.com/Azure/ARO-RP/pkg/util/mocks/dynamichelper"
 )
 
+func TestEsureDeleted(t *testing.T) {
+	ctx := context.Background()
+
+	for _, tt := range []struct {
+		uname     string
+		grpkind   string
+		namespace string
+		name      string
+		wantErr   error
+		grpvr     *schema.GroupVersionResource
+	}{
+		{
+			uname:     "Test EnsureDeleted 1",
+			grpkind:   "configmap",
+			namespace: "test-namespace-1",
+			name:      "test-name-1",
+			wantErr:   errors.New("this is a new error"),
+			grpvr:     &schema.GroupVersionResource{Group: "", Version: "v1", Resource: "configmaps"},
+		},
+		{
+			uname:     "Test EnsureDeleted 2",
+			grpkind:   "configmap",
+			namespace: "test-namespace-2",
+			name:      "test-name-2",
+			wantErr:   errors.New("This is another error"),
+			grpvr:     &schema.GroupVersionResource{Group: "", Version: "v1", Resource: "configmaps"},
+		},
+	} {
+		t.Run(tt.uname, func(t *testing.T) {
+			mockController := gomock.NewController(t)
+			defer mockController.Finish()
+
+			mockCore := mock_dynamichelper.NewMockGVRResolver(mockController)
+			mockCore.
+				EXPECT().
+				Resolve("configmap", "").
```

Comment:
We can use the data from the test case field. This way it will be easier to add test cases for differents kinds.

```suggestion
				Resolve(tt.grpkind, "").
```